### ISSUE TITLE
Proper way to construct URL from URL components (host and port)

### ISSUE THREAD

**Issue by rutsky** [aio-libs/yarl#38]
Lets assume I have host and port numbers of plain HTTP web server, what is the best way to construct proper URL from them?

Looks like default-created URL is considered as relative URL:

```
>>> from yarl import URL
>>> URL().with_host('::1')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/bob/stuff/cscenter/testing/env/lib/python3.5/site-packages/yarl/__init__.py", line 604, in with_host
    raise ValueError("host replacement is not allowed "
ValueError: host replacement is not allowed for relative URLs
>>> 
```

Currently I'm constructing a dummy URL and replacing host and port in it:

```
>>> from yarl import URL
>>> URL('http://example.org').with_host('::1').with_port(8080)
URL('http://::1:8080')
>>> 
```

**Comment by asvetlov** [comment]
Why don't use `URL('http://{}:{}'.format(host, port))` ?

**Comment by rutsky** [comment]
@asvetlov this will not work for example with IPv6 addresses:

```
>>> from yarl import URL
>>> URL('http://{}:{}'.format('::1', 8080))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/bob/stuff/cscenter/testing/env/lib/python3.5/site-packages/yarl/__init__.py", line 162, in __init__
    "Invalid URL: host is required for abolute urls.")
ValueError: Invalid URL: host is required for abolute urls.
>>>
```

Huh, but my example doesn't work either:

```
>>> URL('http://example.org').with_host('::1').with_port(8080)
URL('http://::1:8080')
```

This works as expected:

```
>>> URL('http://example.org').with_host('::1')
URL('http://[::1]')
```

So `with_port()` breaks IPv6 hostname...

**Comment by asvetlov** [comment]
Fixed by #58 I believe.
